hadoop-hdfs-project/hadoop-hdfs-client/src/main/proto/datatransfer.proto

```diff
@@ -38,6 +38,7 @@ message DataTransferEncryptorMessageProto {
     SUCCESS = 0;
     ERROR_UNKNOWN_KEY = 1;
     ERROR = 2;
+    ERROR_ACCESS_TOKEN = 3;
```

Comment:
If the server sends `ERROR_ACCESS_TOKEN` and the client is old (i.e. this patch is not applied), client cannot understand the `ERROR_ACCESS_TOKEN` and it is treated as `SUCCESS (0)`, which is the default value in proto2. In my understanding, setting `SUCCESS` as default was a mistake when creating the enum.

Would you try adding an optional bool for the status?

Reply:
> If the server sends ERROR_ACCESS_TOKEN and the client is old (i.e. this patch is not applied), client cannot understand the ERROR_ACCESS_TOKEN and it is treated as SUCCESS (0),

I see. In such a case, the behavior of the client-side is the same, got some exception in DFSInputStream and long sleep happens (the exception happens will be different though). But I agree it's not graceful, I added accessTokenError bool and set it true with ERROR status. Then old client behaves completely the same.

> In my understanding, setting SUCCESS as default was a mistake when creating the enum.

I agree. I changed the handling of status in readSaslMessage to reject unknown status